core/src/main/java/org/elasticsearch/action/support/nodes/BaseNodesResponse.java

```diff
@@ -91,11 +105,36 @@ public TNodeResponse getAt(int position) {
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         clusterName = ClusterName.readClusterName(in);
+        failures = in.readArray(FailedNodeException[]::new, FailedNodeException::new);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         clusterName.writeTo(out);
+        out.writeArray(failures);
     }
+
+    /**
+     * Expected to be invoked by any {@code BaseNodesResponse} that implements {@code ToXContent} at the top level of the response object.
+     */
+    protected XContentBuilder toInnerXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
> Note that for the broadcast shards header in the response we use _shards. I think it should be the same here so _nodes?

Convincing enough to me. I glanced around and a few of its uses did add a `"shards"` field (though not always at the same level).